### DISCUSSION TITLE
Run keycloak 19 in docker

### DISCUSSION THREAD

**Discussion by HoussemGargouri** [keycloak/keycloak#13373] [Archive]
Hello,

When i want to run keycloak 19 in docker with the ENTRYPOINT in dockerfine `--db=postgres` i had an error : 

`Unknown option: '--db'
Possible solutions: --db-url, --db-url-host, --db-url-database, --db-url-port, --db-url-properties, --db-username, --db-password, --db-schema, --db-pool-initial-size, --db-pool-min-size, --db-pool-max-size
Try 'kc.sh start --help' for more information on the available options.
`

And When i delete this variable from the dockerfile i had another error : 

`"ERROR: No suitable driver found for jdbc:postgresql://<mydatabase>:5432/database"`

In the documentation this variable still exist. But I don't understand why I got the error. Any idea please ? Thanks !

**Comment by tintin92350** [comment]
Hi,

I have the same issue when I try to run keycloak 19. 

But I think that `--db` option is only available for building command and not starting command. see [official keycloak how to run in container](https://www.keycloak.org/server/containers). 

As you can notice, they set `KC_DB` (an environment variable alternative of `--db` cli option)

We can see other clues in the [First step: Run a build explicitly of official keycloak website](https://www.keycloak.org/server/configuration) when they use the cli option with building process.

I have tried to build my keycloak thanks to their example and it seems to work!

Here is my Dockerfile:

```
FROM quay.io/keycloak/keycloak:19.0.0 as builder

ENV KC_HEALTH_ENABLED=true
ENV KC_METRICS_ENABLED=true
ENV KC_FEATURES=token-exchange
ENV KC_DB=postgres

RUN curl -sL https://github.com/aerogear/keycloak-metrics-spi/releases/download/2.5.3/keycloak-metrics-spi-2.5.3.jar -o /opt/keycloak/providers/keycloak-metrics-spi-2.5.3.jar
RUN /opt/keycloak/bin/kc.sh build

FROM quay.io/keycloak/keycloak:19.0.0
COPY --from=builder /opt/keycloak/ /opt/keycloak/
WORKDIR /opt/keycloak

ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start --proxy=edge", "--optimized"]
```

(It comes with metrics as health but you can remove them)

**Reply by HoussemGargouri** [reply]
It worked when i use the `ENV KC_DB=postgres` in building time thx !

**Reply by toudidel** [reply]
`--db` is used in official documentation on page https://www.keycloak.org/server/containers in section _Running a standard keycloak container_
